### PR TITLE
Feature/new route redeem code

### PR DESCRIPTION
**Pull request by sunilsabatp**
to do -> create a new route to handle redeem

steps: 
1.use the route `/profile/redeem/:code` to see the result

### CODE REVIEW

**Comment by sagararyal** [comment]
#### 📝 Changed routes:


| [**/profile/redeem/\[code\]**](https://planet-webapp-bahoyn160-planetapp.vercel.app/profile/redeem/[code]) |
|:-:|
|<a href="https://planet-webapp-bahoyn160-planetapp.vercel.app/profile/redeem/[code]">Screenshot of /profile/redeem/[code]</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/1tlbYp2akt0DKtj4abjybM1unjdl8ymAS4ZP8zpcEHj88-8qcU4BN4jAA3MaxdoasWWIP6tlnM0x-I_hEriHxA.png">(view full size)</a> |







Commit 40bd3e29371211c74fd2f98fd9e27e5cb4e56706 (https://planet-webapp-bahoyn160-planetapp.vercel.app).

**Review by mariahosfeld — CHANGES_REQUESTED** [review]
Route works, but the following changes are needed:
- [x] Make sure that new tree count is shown automatically, currently a refresh is needed
- [ ] Provide option to redeem another code in each modal (successfully, already redeemed, invalid)

**Comment by sunilsabatp** [comment]
> Route works, but the following changes are needed:
> 
> * [ ]  Make sure that new tree count is shown automatically, currently a refresh is needed
> * [ ]  Provide option to redeem another code in each modal (successfully, already redeemed, invalid)

the first issue have been fixed in one of my pr  i.e redeem end point

**Comment by mariahosfeld** [comment]
Functionality works for me

**Comment by norbertschuler** [comment]
If this PR replaces #1554 (which I then just unnecessarily tested :-() please close the outdated PR.

**Comment by mohitb35** [comment]
> If this PR replaces #1554 (which I then just unnecessarily tested :-() please close the outdated PR.

@norbertschuler This PR is in addition to #1554 - #1554 updates the redemption process from the profile via the RedeemModal, and this PR focuses on a route to directly redeem a code.

**Comment by norbertschuler** [comment]
I would prefer upgrading `package-lock.json` only in the PR has changed `package.json` or do upgrades in separate PR's which then check if all modules still work well.

File: pages/claim/[type]/[code].tsx [hunk 1]
```diff
@@ -1,97 +1,60 @@
...
-      } else {
         setCode(router.query.code);
-        setType(router.query.type);
-        setRouterReady(true);
+      } else {
         setErrorMessage('');
       }
     }
-  }, [router]);
+  }, [router, router.query.type, ready]);
```
Comment (prachigarg19) [inline pages/claim/[type]/[code].tsx:42]: Why do we need router and router.query.type both as dependencies here?

**Review by prachigarg19 — APPROVED** [review]

**Comment by mariahosfeld** [comment]
> * [ ]  Using some random url as https://planet-webapp-hxl3eqtui-planetapp.vercel.app/claim/test/xxx-xxx results into some endless spinner (wasn't working well before):
> 
> 
> Bildschirmfoto 2022-10-03 um 12 15 11

This has been solved, will now result in "invalid code" message.

**Pull request merged**